crates/bevy_ecs/src/query/access.rs

```diff
@@ -2,6 +2,10 @@ use crate::storage::SparseSetIndex;
 use fixedbitset::FixedBitSet;
 use std::marker::PhantomData;
 
+/// `Access` keeps track of read and write accesses to [`SparseSetIndices`](SparseSetIndex).
+///
+/// This type is useful for calculating whether two different `Access` instances collide and what
+/// their collisions are.
```

Comment:
This plural has the same problem, so '[`SparseSetIndex`] implementors' might be more precise. However, in this case I feel like the `SparseSetIndex` bound is a red herring in this struct, and it should just be removed in favour of usize everywhere.

That is
```suggestion
/// `Access` keeps track of read and write accesses to values within a collection.
/// This is used for ensuring systems are executed soundly.
```

However, actually removing this bound should be seperately discussed, probably in another PR.